common/src/main/java/com/revenuecat/purchases/common/CustomerInfoFactory.kt

```diff
@@ -0,0 +1,80 @@
+package com.revenuecat.purchases.common
+
+import android.net.Uri
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.EntitlementInfos
+import com.revenuecat.purchases.VerificationResult
+import com.revenuecat.purchases.common.networking.HTTPResult
+import com.revenuecat.purchases.utils.Iso8601Utils
+import com.revenuecat.purchases.utils.optNullableString
+import com.revenuecat.purchases.utils.parseExpirations
+import com.revenuecat.purchases.utils.parsePurchaseDates
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.Collections.emptyMap
+
+/**
+ * Builds a CustomerInfo
+ * @throws [JSONException] If the json is invalid.
+ */
+object CustomerInfoFactory {
```

Comment:
This change caused most of the other changes on this PR. I thought making the CustomerInfo construction be a part of JSONObject didn't make sense anymore when there is more data we need now. I thought about adding the verification result to the json, but that didn't make sense to me since that json is supposed to be the response body from the backend and I didn't want to mix data. I think this way is also more extensible in case creating a CustomerInfo changes in the future.